PurchasesCoreSwift/Misc/SystemInfo.swift

```diff
@@ -0,0 +1,169 @@
+    @objc public static var isSandbox: Bool {
+        let url = Bundle.main.appStoreReceiptURL
+        guard let url = url else {
+            return false
+        }
+
+        let receiptURLString = url.path
+        return receiptURLString.contains("sandboxReceipt")
+    }
+    @objc public static var frameworkVersion: String { // TODO: automate the setting of this, if it hasn't been.
+        return "3.12.0-SNAPSHOT"
+    }
+
+    @objc public static var systemVersion: String {
+        return ProcessInfo().operatingSystemVersionString
+    }
+
+    @objc public static var appVersion: String {
+        return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+    }
+
+    @objc public static var buildVersion: String {
+        return Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? ""
+    }
+
+    @objc public static var platformHeader: String {
+        return Self.forceUniversalAppStore ? "iOS" : platformHeaderConstant
+    }
+
+    @objc public static var identifierForVendor: String? {
+        #if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+            return UIDevice.current.identifierForVendor?.uuidString
+        #elseif os(watchOS)
+            return WKInterfaceDevice.current().identifierForVendor?.uuidString
+        #else
+            return nil
+        #endif
+    }
+
+    private static var privateProxyURL: URL?
+    @objc public static var proxyURL: URL? {
+        get {
+            return privateProxyURL
+        }
+        set {
+            privateProxyURL = newValue
+            if let privateProxyURLString = newValue?.absoluteString {
+                Logger.info(Strings.configure.configuring_purchases_proxy_url_set
+                                .replacingOccurrences(of: "%@", with: privateProxyURLString))
+            }
+        }
+    }
+
+    static var defaultServerHostURL: URL {
+        return URL(string: defaultServerHostName)!
+    }
+
+    @objc public static var serverHostURL: URL {
+        return self.proxyURL ?? Self.defaultServerHostURL
+    }
+
+    private var isApplicationBackgrounded: Bool {
+    #if os(iOS) || (targetEnvironment(simulator) && os(iOS))
+        return self.isApplicationBackgroundedIOS
+    #elseif os(tvOS)
+        return UIApplication.shared.applicationState == UIApplication.State.background
+    #elseif os(macOS)
+        return false
+    #elseif os(watchOS)
+        return  WKExtension.shared().applicationState == WKApplicationState.background
+    #endif
+    }
+
+    #if os(iOS)
+    // iOS App extensions can't access UIApplication.sharedApplication, and will fail to compile if any calls to
+    // it are made. There are no pre-processor macros available to check if the code is running in an app extension,
+    // so we check if we're running in an app extension at runtime, and if not, we use KVC to call sharedApplication.
+    @objc private var isApplicationBackgroundedIOS: Bool {
```

Comment:
if we have stuff that's private, I don't think we should have them be `objc` - we're not getting any value out of it, but we're adding restrictions for the types that can be used, in a codebase that will soon be swift-only